_posts/2019-02-20-lecture-notes-template.md

```diff
@@ -0,0 +1,361 @@
+P_{t+1 \vert t} &= E((X_{t+1} - \hat X_{t+1 \vert t})(X_{t+1} - \hat X_{t+1 \vert t})^T \vert Y_1...Y_t) \\
+&= E((AX_t + GW_t - \hat X_{t+1 \vert t})(AX_t + GW_t - \hat X_{t+1 \vert t})^T \vert Y_1...Y_t) \\
+&= AP_{t \vert t}A + GQG^T
+\end{aligned}
+</d-math>
+
+For the observation model, which is defined as $Y_{t} = CX_t + \nu_t$, where $\nu \sim \mathcal{N}(0,R)$, we can compute the mean and variance as follows:
+<d-math block>
+\begin{aligned}
+E(Y_{t+1} \vert Y_1...Y_t ) &= E(CX_{t+1} + \nu_{t+1} \vert Y_1...Y_t ) = C\hat X_{t+1|t} \\
+E((Y_{t+1} - \hat Y_{t+1 \vert t})(Y_{t+1} - \hat Y_{t+1 \vert t})^T \vert Y_1...Y_t) &= CP_{t+1|t}C^T + R\\
+\end{aligned}
+</d-math>
+
+Finally, we can also derive the following:
+$$ E((Y_{t+1} - \hat Y_{t+1 \vert t})(X_{t+1} - \hat X_{t+1 \vert t})^T \vert Y_1...Y_t) = CP_{t+1|t} $$
+
+### Update Step Derivation
+Recall that for two gaussian distributions $X_1, X_2$, their joint distribution is a Gaussian with the following mean and variance:
+<d-math block>
+\begin{aligned}
+\mu &=
+  \begin{bmatrix}
+    \mu_1 \\
+    \mu_2
+  \end{bmatrix} \\
+\sum &=
+  \begin{bmatrix}
+    \sum_{11} & \sum_{12} \\
+    \sum_{21} & \sum_{22}
+  \end{bmatrix}
+\end{aligned}
+</d-math>
+Using this result in combination with the derivations from the predict step, we can deduce that $P(X_{t+1}, Y_{t+1} \vert Y_1...Y_t) \sim \mathcal{N}(M_{t+1}, V_{t+1})$, where
+<d-math block>
+\begin{aligned}
+M_{t+1} &=
+  \begin{bmatrix}
+    \hat X_{t+1|t} \\
+    C\hat X_{t+1|t}
+  \end{bmatrix} \\
+V_{t+1} &=
+  \begin{bmatrix}
+    P_{t+1|t} & P_{t+1|t}C^T \\
+    CP_{t+1|t} & CP_{t+1|t}C^T + R
+  \end{bmatrix}
+\end{aligned}
+</d-math>
+
+Given a joint distribution of two Gaussian random variables $X_1, X_2$, recall that their conditional and marginal probability distributions are defined as follows:\\
+$$ P(X_2) = \mathcal{N}(M_2^m, V_2^m) $$ \\
+$$ M_2^m = \mu_2 $$ \\
+$$ V_2^m = \sum_{22} $$
+
+$$ P(X_{1 \vert 2}) = \mathcal{N}(M_{1 \vert 2}^m, V_{1 \vert 2}^m) $$ \\
+$$ M_{1 \vert 2}^m = \mu_1 + \sum_{12} \sum_{22}^{-1} (X_2 - \mu_2) $$ \\
+$$ V_{1 \vert 2}^m = \sum_{11} - \sum_{12} \sum_{22}^{-1} \sum_{21} $$
+
+Using this information, we can compute the measurement update (update step) as:
+<d-math block>
+\begin{aligned}
+\hat X_{t+1 \vert t+1} &= \hat X_{t+1 \vert t} + K_{t+1} (Y_{t+1} - C \hat X_{t+1 \vert t}) \\
+P_{t+1 \vert t+1} &= P_{t+1 \vert t} - KCP_{t+1 \vert t}\\
+\end{aligned}
+</d-math>
+Here $K_{t+1} = P_{t+1 \vert t} C^T (C P_{t+1 \vert t} C^T + R)^{-1}$ and is known as the Kalman gain matrix. An interesting point to note is that $K_t$ is independent of $Y_t$ (i.e. the data), so it can be precomputed to improve efficiency.
+
+### Example of Kalman Filtering in 1-D
+Suppose we have some noisy observations of a particle performing a random walk in 1-D. Assume our states and observations are defined as follows: \\
+$ X_{t \vert t-1} = X_{t-1} + W $ where $W \sim \mathcal{N}(0,\sigma_x)$ \\
```

Comment:
please use \begin{align} and center the equations

Reply:
done